declarations/custom.d.ts

```diff
@@ -19,6 +19,7 @@ interface Game {
 interface Room {
     basicMatrix: CostMatrix;
     findStructures<T>(structureType: string): T[];
+    findStructures<T>(structureType: string): Structure[];
```

Comment:
In order to limit the result to type Structure, I suggest simply changing the previous line to `findStructures<T extends Structure>(structureType: string): T[];`. This will still allow to be more specific with type T if required.

Reply:
That seems good to me too, I think I tried to do this once but it was on the prototype assignment rather than the interface declaration and so I wasn't sure why it wouldn't work

Reply:
I'm new to TS and didn't want to try a trickier solution like that. Looks like it would work, though.